### PR TITLE
feat(swr): add datasource SWR shared repositories

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
add datasource SWR shared repositories

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```
add datasource SWR shared repositories
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/swr" TESTARGS="-run TestAccDataSourceSharedRepositories_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/swr -v -run TestAccDataSourceSharedRepositories_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceSharedRepositories_basic
=== PAUSE TestAccDataSourceSharedRepositories_basic
=== CONT  TestAccDataSourceSharedRepositories_basic
--- PASS: TestAccDataSourceSharedRepositories_basic (69.26s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/swr       69.326s
```

### CODE REVIEW

File: docs/data-sources/swr_shared_repositories.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+* `organization` - (Optional, String) Specifies the name of the organization (namespace) the repository belongs.
+
+* `name` - (Optional, String) Specifies the name of the repository.
+
+* `center` - (Required, String) Specifies the center of the repository.
+  + **self:** - Indicates my shared images.
+  + **thirdparty:** - Indicates images shared by others with me.
+
+* `domain_name` - (Optional, String) Specifies The account name of the repository owner.
```
Comment (saf3dfsa) [inline docs/data-sources/swr_shared_repositories.md]:
```suggestion
* `domain_name` - (Optional, String) Specifies the account name of the repository owner.
```

File: huaweicloud/services/acceptance/swr/data_source_huaweicloud_swr_shared_repositories_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+func testAccDataSourceSharedRepositories_basic(rName string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_swr_shared_repositories" "test" {
+  depends_on = [huaweicloud_swr_repository_sharing.test]
+  
+  center     = "self"
```
Comment (saf3dfsa) [inline huaweicloud/services/acceptance/swr/data_source_huaweicloud_swr_shared_repositories_test.go]:
```suggestion
  center = "self"
```

File: huaweicloud/services/swr/data_source_huaweicloud_swr_shared_repositories.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+	listSharedRepositoriesHttpUrl := "v2/manage/shared-repositories"
+	listSharedRepositoriesPath := client.Endpoint + listSharedRepositoriesHttpUrl
+	listSharedRepositoriesQueryParams := buildListSharedRepositoriesQueryParams(d)
+	listSharedRepositoriesPath += listSharedRepositoriesQueryParams
+
+	listSharedRepositoriesOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
```
Comment (saf3dfsa) [inline huaweicloud/services/swr/data_source_huaweicloud_swr_shared_repositories.go]: The default okcode includes 200, you can remove this

File: docs/data-sources/swr_shared_repositories.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+* `center` - (Required, String) Specifies the center of the repository.
+  + **self:** - Indicates my shared images.
+  + **thirdparty:** - Indicates images shared by others with me.
+
+* `organization` - (Optional, String) Specifies the name of the organization (namespace) the repository belongs.
+
+* `name` - (Optional, String) Specifies the name of the repository.
+
+* `domain_name` - (Optional, String) Specifies the account name of the repository owner.
+
```
Comment (houpeng80) [inline docs/data-sources/swr_shared_repositories.md:37]: status should be added to query param

File: docs/data-sources/swr_shared_repositories.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+<a name="swr_shared_repositories"></a>
+The `repositories` block supports:
+
+* `organization` - The name of the organization (namespace) the repository belongs.
+
+* `name` - The name of the repository.
+
+* `domain_name` - The account name of the repository owner.
+
+* `status` - Whether this repository is shared with others.
```
Comment (houpeng80) [inline docs/data-sources/swr_shared_repositories.md]: the description is not suitable, it indicates whether the sharing has expired

File: huaweicloud/services/swr/data_source_huaweicloud_swr_shared_repositories.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+func flattenSharedRepositoriesResponse(rawParams []interface{}, d *schema.ResourceData) []interface{} {
+	if len(rawParams) == 0 {
+		return nil
+	}
+	results := make([]interface{}, 0)
+	for _, v := range rawParams {
+		name := utils.PathSearch("name", v, "").(string)
+		if val, ok := d.GetOk("name"); ok && name != val {
+			continue
+		}
```
Comment (houpeng80) [inline huaweicloud/services/swr/data_source_huaweicloud_swr_shared_repositories.go:216]: the query params have contains name, why use the name to filter the result angin?

**Pull request merged**